src/cpp/library/string

```diff
@@ -1708,14 +1695,14 @@ __ESBMC_HIDE:
 
 size_t string::find(char c, size_t pos) const
 {
-__ESBMC_HIDE:
-  __ESBMC_assert((pos >= 0) && (pos < this->length()), "string overflow");
-  __ESBMC_assert(c != NULL, "The parameter must to be different than NULL");
-  int i;
-  int len = this->length();
-  for (i = 0; i < len; i++)
+  __ESBMC_HIDE:
+  __ESBMC_assert(pos < this->length(), "string overflow");
```

Comment:
Isn't this assert redundant with the bounds check? Additionaly [CPP Reference](https://cplusplus.com/reference/string/string/find/) states for pos: `If this is greater than the string length, the function never finds matches.`

Reply:
For this PR, I'll leave the assertions as they are now; we can revise these pre-conditions in all our C++ OMs in another PR.